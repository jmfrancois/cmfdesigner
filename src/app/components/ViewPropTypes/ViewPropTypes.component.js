import React from 'react';
import PropTypes from 'prop-types';
import { cmfConnect } from '@talend/react-cmf';

function renderPropType(key, obj) {
	return (
		<tr key={key}>
			<td>{key}</td>
			<td>{obj[key].type}</td>
			<td>{obj[key].required ? 'true' : 'false'}</td>
		</tr>
	);
}

function ViewPropTypes(props) {
	if (!props.data) {
		return null;
	}

	return (
		<table className="table table-bordered">
			<thead>
				<tr>
					<th>Key</th>
					<th>Type</th>
					<th>isRequired</th>
				</tr>
			</thead>
			<tbody>
				{Object.keys(props.data).map(key => renderPropType(key, props.data))}
			</tbody>
		</table>
	);
}

ViewPropTypes.displayName = 'ViewPropTypes';
ViewPropTypes.propTypes = {
	data: PropTypes.object,
};

export default cmfConnect({})(ViewPropTypes);
