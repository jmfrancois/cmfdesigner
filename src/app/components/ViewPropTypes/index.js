import ViewPropTypes from './ViewPropTypes.component';

export default ViewPropTypes;
