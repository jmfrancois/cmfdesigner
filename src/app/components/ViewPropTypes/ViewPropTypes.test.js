import React from 'react';
import { shallow } from 'enzyme';

import ViewPropTypes from './ViewPropTypes.component';

describe('ViewPropTypes', () => {
	it('should render', () => {
		const wrapper = shallow(
			<ViewPropTypes.WrappedComponent data={{}} />
		);
		expect(wrapper.getElement()).toMatchSnapshot();
	});
});
