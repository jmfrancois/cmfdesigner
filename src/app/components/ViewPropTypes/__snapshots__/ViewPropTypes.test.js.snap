// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ViewPropTypes should render 1`] = `
<table
  className="table table-bordered"
>
  <thead>
    <tr>
      <th>
        Key
      </th>
      <th>
        Type
      </th>
      <th>
        isRequired
      </th>
    </tr>
  </thead>
  <tbody>
    
  </tbody>
</table>
`;
